enhancements/fbc-run-bundle.md

```diff
@@ -0,0 +1,247 @@
+---
+title: fbc-run-bundle
+authors:
+  - "@rashmigottipati"
+reviewers:
+  - "@joelanford"
+  - "@jmrodri"
+  - "@vramaraju"
+approvers:
+  - "@joelanford"
+  - "@jmrodri"
+  - "@vramaraju"
+creation-date: 2022-04-20
+last-updated: 2022-04-20
+status: implementable
+see-also:
+replaces:
+superseded-by:
+
+---
+
+# Add FBC support to run bundle
+
+## Release Signoff Checklist
+
+- [X] Enhancement is `implementable`
+- [X] Design details are appropriately documented from clear requirements
+- [X] Test plan is defined
+- [X] Graduation criteria for dev preview, tech preview, GA
+
+## Open Questions
+
+- Do we need to build a File-Based Catalog to run this command?
+
+  - We can use `quay.io/operatorhubio/catalog:latest` for testing and generate a File-Based Catalog at runtime when the run bundle command is invoked.
+
+- Do we need to make our own image to use as a default one for run bundle and run bundle-upgrade?
+
+  - We might have to build an image similar to `quay.io/operatorframework/catalog:latest` which uses opm as the base image and also has a built catalog in it.
```

Comment:
Opm has not its own base image to create catalogs from scratch?

Reply:
We are using `quay.io/operator-framework/opm:latest` as the base index image for `run bundle`, but it does not have any catalog with it. I think there's `quay.io/operatorframework/catalog:latest` which has opm as a base image and a minimum catalog built with the image. 

I wonder if we should be making a similar image for `run bundle` for usage in Operator SDK.

Reply:
Why we would need to create the image?
What would be the pros and cons?
What are the motivations?

Reply:
What would the minimum catalog contain? In the existing `run bundle` with SQLite, the default image (if one isn't provided) is already using `quay.io/operator-framework/opm:latest`, which does not contain a catalog. Why would an FBC-based execution need the base image to contain a catalog when the SQLite execution does not?

Reply:
That makes sense, thanks for clarifying @joelanford. 

My original intention was that we need a veneer to build off of (something you mentioned in the bundle-upgrade design discussion and use that to construct an actual FBC. A minimum catalog with one `olm.bundle` blob, one `olm.channel` blob, and one `olm.package` blob.

But since we already do that as part of creation of FBC on the fly, I don't think we need an image that has opm and a minimum FBC. `quay.io/operator-framework/opm:latest` should suffice for both index formats.